### PR TITLE
Mocksolver improvements

### PR DESCRIPTION
**Pull request by joaquimg**

### CODE REVIEW

**Comment by odow** [comment]
Does this supersede #26?

**Comment by joaquimg** [comment]
I just added the two vector for xpress, so #26 idea was completely dropped.
This add tests for quadratics SOS and some more.
And now fixes 1 indexing.
We should merge asap.

File: src/constraints/scalaraffine.jl [hunk 1]
```diff
@@ -154,7 +154,7 @@ function MOI.get(m::LinQuadOptimizer, ::MOI.ConstraintFunction, c::LCI{<: LinSet
     # TODO (@joaquim) fix indexing
     terms = map(
         (v,c)->MOI.ScalarAffineTerm{Float64}(c,v),
-        m.variable_references[colidx+1],
+        m.variable_references[colidx],
```
Comment (joaquimg) [inline src/constraints/scalaraffine.jl]: @odow 1 indexing here, need to fix GLPK and Gurobi

Comment (odow) [reply src/constraints/scalaraffine.jl]: Cool. I knew there were probably things like this. It gets too confusing switching between all the different formats and 0- and 1-indexing.

File: src/constraints/scalaraffine.jl [hunk 2]
```diff
@@ -151,9 +151,10 @@ MOI.canget(m::LinQuadOptimizer, ::MOI.ConstraintFunction, ::Type{<:LCI{<: LinSet
 function MOI.get(m::LinQuadOptimizer, ::MOI.ConstraintFunction, c::LCI{<: LinSets})
     # TODO more efficiently
     colidx, coefs = get_linear_constraint(m, m[c])
+    # TODO (@joaquim) fix indexing
```
Comment (odow) [inline src/constraints/scalaraffine.jl]: Is this still a todo?

**Comment by odow** [comment]
LGTM. You can squash and merge when you're happy.

File: src/constraints/scalaraffine.jl (hunk 2 above)
Comment (joaquimg) [reply src/constraints/scalaraffine.jl]: nope, will fix

**Pull request merged**